pkg/api/v1/clients/kube/cache/cache.go

```diff
@@ -136,114 +155,138 @@ func NewKubeCoreCacheWithOptions(ctx context.Context, client kubernetes.Interfac
-				return client.CoreV1().Services(nsToWatch).List(ctx, options)
-			}
-			informer := skkube.NewSharedInformer(typeCtx, resyncDuration, &v1.Service{}, list, watch)
-			informers = append(informers, informer)
-			lister := kubelisters.NewServiceLister(informer.GetIndexer())
-			services[nsToWatch] = lister
-		}
-		{
-			var typeCtx = nsCtx
-			if ctxWithTags, err := tag.New(nsCtx, tag.Insert(skkube.KeyKind, "ConfigMap")); err == nil {
-				typeCtx = ctxWithTags
-			}
-			// ConfigMap
-			watch := client.CoreV1().ConfigMaps(nsToWatch).Watch
-			list := func(options metav1.ListOptions) (runtime.Object, error) {
-				return client.CoreV1().ConfigMaps(nsToWatch).List(ctx, options)
-			}
-			informer := skkube.NewSharedInformer(typeCtx, resyncDuration, &v1.ConfigMap{}, list, watch)
-			informers = append(informers, informer)
-			lister := kubelisters.NewConfigMapLister(informer.GetIndexer())
-			configMaps[nsToWatch] = lister
-		}
-		{
-			var typeCtx = nsCtx
-			if ctxWithTags, err := tag.New(nsCtx, tag.Insert(skkube.KeyKind, "Secrets")); err == nil {
-				typeCtx = ctxWithTags
-			}
-			// Secrets
-			watch := client.CoreV1().Secrets(nsToWatch).Watch
-			list := func(options metav1.ListOptions) (runtime.Object, error) {
-				return client.CoreV1().Secrets(nsToWatch).List(ctx, options)
-			}
-			informer := skkube.NewSharedInformer(typeCtx, resyncDuration, &v1.Secret{}, list, watch)
-			informers = append(informers, informer)
-			lister := kubelisters.NewSecretLister(informer.GetIndexer())
-			secrets[nsToWatch] = lister
-		}
-
-	}
-
-	var namespaceLister kubelisters.NamespaceLister
-	if len(namesapcesToWatch) == 1 && namesapcesToWatch[0] == metav1.NamespaceAll {
-
-		// Pods
-		watch := client.CoreV1().Namespaces().Watch
-		list := func(options metav1.ListOptions) (runtime.Object, error) {
-			return client.CoreV1().Namespaces().List(ctx, options)
-		}
-		nsCtx := ctx
-		if ctxWithTags, err := tag.New(nsCtx, tag.Insert(skkube.KeyNamespaceKind, skkube.NotEmptyValue(metav1.NamespaceAll)), tag.Insert(skkube.KeyKind, "Namespaces")); err == nil {
-			nsCtx = ctxWithTags
-		}
-		informer := skkube.NewSharedInformer(nsCtx, resyncDuration, &v1.Namespace{}, list, watch)
-		informers = append(informers, informer)
-		namespaceLister = kubelisters.NewNamespaceLister(informer.GetIndexer())
-	}
-
 	k := &kubeCoreCaches{
 		podListers:       pods,
 		serviceListers:   services,
 		configMapListers: configMaps,
 		secretListers:    secrets,
-		namespaceLister:  namespaceLister,
+		client:           client,
+		ctx:              ctx,
+		resyncDuration:   resyncDuration,
+		informers:        informers,
+	}
+
+	for _, nsToWatch := range namesapcesToWatch {
+		k.addNewNamespace(nsToWatch)
 	}
 
-	kubeController := controller.NewController("kube-plugin-controller",
-		controller.NewLockingSyncHandler(k.updatedOccured), informers...,
+	// since we now allow namespaces to be registered dynamically we will always create the namespace lister
+	k.addNamespaceLister()
```

Comment:
Always create the namepsace lister. Was thinking that I might get a flag, but I don't think it is needed.

Reply:
I don't think a flag is necessary

Reply:
Flag is necessary for clients.